standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/CompactionMetricData.java

```diff
@@ -45,6 +45,8 @@ final class CompactionMetricData {
 
   private Map<String, Long> stateCount;
 
+  private Map<String, Integer> poolCount;
```

Comment:
I wonder if the counter can be Long or not. I mean, the `CC_ID` column in the meta-store database is big int. I don't know our usage statistics but I wonder: it is possible to have so many compaction requests in pool that Integer is not enough?

Reply:
That is totally impossible, compaction related queries would start to fail much-much earlier. Even a few thousand unprocessed compaction entries mean a serious issue and start affecting performance.

Reply:
Great, thank you.